src/components/Select/MultiSelect/hooks.js

```diff
@@ -0,0 +1,167 @@
+import { useState, useRef, useCallback, useEffect } from "react";
+import { useClickOutside, useHandleKeyboardHook, usePrevious } from "../hooks";
+
+export function useMultiSelect(
+  disabled,
+  autoFocus,
+  onChange,
+  options,
+  defaultValues,
+  getOptionLabel,
+  getOptionValue,
+  filterOptions
+) {
+  const optionContainerRef = useRef(null);
+  const dropdownRef = useRef(null);
+
+  const [menuOpened, setMenuOpened] = useState(false);
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState(0);
+
+  const _options = filterOptions ? options.filter(filterOptions) : options;
+
+  const defaultValueKeyGetter = useCallback(({ value }) => value, []);
+  const defaultLabelKeyGetter = useCallback(({ label }) => label, []);
+
+  const getValueKey = getOptionValue || defaultValueKeyGetter;
+  const getLabelKey = getOptionLabel || defaultLabelKeyGetter;
+
+  const [selectedOptions, setSelectedOptions] = useState(
+    _options.filter((_option) =>
+      defaultValues.find(
+        (defaultValue) => getValueKey(defaultValue) === getValueKey(_option)
+      )
+    ) || []
+  );
+
+  useEffect(() => {
+    if (disabled) {
+      setMenuOpened(false);
+      return;
+    }
+    if (autoFocus) setMenuOpened(true);
+  }, [disabled, autoFocus]);
+
+  const previousSelectedOptions = usePrevious(selectedOptions);
+
+  useEffect(() => {
+    if (disabled) return;
+    if (filterOptions && previousSelectedOptions !== selectedOptions)
+      setSelectedOptions(selectedOptions.filter(filterOptions));
+  }, [disabled, selectedOptions, filterOptions, previousSelectedOptions]);
+
+  const removeSelectedOption = (option) => {
+    if (disabled) return;
+    setSelectedOptions(
+      selectedOptions.filter(
+        (selectedOption) => getLabelKey(selectedOption) !== getLabelKey(option)
+      )
+    );
+  };
+
+  const setOption = (option, knownIndex) => {
+    const index =
+      knownIndex ||
+      _options.findIndex(
+        (opt) =>
+          getValueKey(opt) === getValueKey(option) &&
+          getLabelKey(opt) === getLabelKey(option)
+      );
+    if (
+      selectedOptions.find(
+        (selectedOption) => getLabelKey(selectedOption) === getLabelKey(option)
+      )
```

Comment:
This `find` method is used many times. I think you should consider moving it to a helper and import it. Also, I think it's always better if we declare a variable and use it as the condition.